http.go

```diff
@@ -9,13 +9,14 @@ import (
 	"strings"
 	"time"
 
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/apis/options"
 	"github.com/oauth2-proxy/oauth2-proxy/pkg/logger"
 )
 
 // Server represents an HTTP server
 type Server struct {
 	Handler http.Handler
-	Opts    *Options
+	Opts    *options.Options
```

Comment:
This is not idiomatic Go: https://blog.golang.org/package-names ("Avoid stutter")

Reply:
I appreciate that and eventually will rename them. For now I'm trying to create a minimal diff and make baby steps towards a longer goal of moving all of the external configuration content into one package and then breaking it down into smaller structures and eventually, adding a structured configuration to replace our flat config as it is now.

Reply:
`options.Options` is correct to me. I agree that stutter should be avoided, using `Options` is clearer than any alternative I can think of.